docs/lang/articles/advanced/differentiable_programming.md

```diff
@@ -296,41 +302,21 @@ func_break_rule_2.grad()
 assert x.grad[1] == 4.0
 assert x.grad[2] == 3.0
 ```
-### Kernel Simplicity Rule
 
-:::note Kernel Simplicity Rule
-Kernel body must consist of multiple simply nested for-loops. For example, each for-loop can either contain exactly one (nested) for-loop (and no other statements), or a group of statements without loops.
+### Avoid mixed usage of parallel for-loop and non-for statements
+Mixed usage of parallel for-loops and non-for statements are not supported in the autodiff system.
```

Comment:
Maybe also mention that using mixed parallel for-loop and non-for stmts will trigger an error? (so that users know they don't get silent erros :P )